website/docs/setup/guides/schemas.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Managing Schemas in Vector
+sidebar_label: Managing Schemas
+description: Learn how to manage log schemas with Vector.
+---
+
+Data comes in all shapes and sizes. Vector has an array (let's call it a vector 😎) of composable functionality for
+decoding your events in the right format, transforming them into the right shape, and passing that data on downstream.
+
+## Defining Global Field Names
+
+> TODO: Not implemented, yet!
+
+By default, Vector primarily operates on three fields: `host`, `message`, and `timestamp`.
+
+```json
+{
+  "host": "my.host.com",
+  "message": "<13>Feb 13 20:07:26 74794bfb6795 root[8539]: i am foobar",
+  "timestamp": "2019-11-01T21:15:47+00:00"
+}
+```
+
+It may be that your data does not follow this convention. In this case you can modify the global defaults for all incoming data in the `log_schema` section of your `config.toml`.
+
+```toml
+[log_schema]
+host_key = "instance" # default "host"
+message_key = "info" # default "message"
+timestamp_key = "datetime" # default "timestamp"
+
+# Sources, transforms, and sinks...
+```
+
+> **Gotcha:** Not all sources use the `host` field.
+>
+> **Gotcha:** The timestamp and message fields are sometimes derived from the source itself. Eg for the file source `timestamp` is the exact time the event was injested, and not configurable.
+
+### Example: Custom timestamp field
+
+Some services will produce logs with the timestamp field mapped to `@timestamp` or some other value.
+
+If your vector pipeline is only consuming data from these sources, you can add the following to your `config.toml`:
+
+```toml
+# TODO: A simple config and a one-liner invocation.
+[log_schema]
+timestamp_key = "@timestamp"
+
+[sources.my_naming_confused_source]
+  type = "logplex"
+  address = "0.0.0.0:8088"
+```
+
+
+
+## Field filtering
+
+> TODO: https://github.com/timberio/vector/issues/1448 is exploring how to make this better.
+
+Sometimes it is advantageous to filter out specific fields during the pipeline. You can use a `remove_fields` transform transform to do this.
```

Comment:
Right, my only concern with using a schema to whitelist fields is the additional (and more complex) effort to define types. But it makes sense! And it is something I'd prefer users do because mixed types can cause all kinds of weird issues.

So I vote that we start with a `schema` transform and implement a whitelist of field names if users still need it.